packages/grafana-runtime/src/types/i18n.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Hook type for translation function that takes an ID, default message, and optional values
+ * @returns A function that returns the translated string
+ */
+type UseTranslateHook = () => (id: string, defaultMessage: string, values?: Record<string, unknown>) => string;
+
+/**
+ * Type for children elements in Trans component
+ * Can be either React nodes or an object of values
+ */
+type TransChild = React.ReactNode | Record<string, unknown>;
+
+/**
+ * Props interface for the Trans component used for internationalization
+ * @interface
+ * @property {string} i18nKey - The translation key to look up
+ * @property {TransChild | readonly TransChild[]} [children] - Child elements or values to interpolate
+ * @property {readonly React.ReactElement[] | { readonly [tagName: string]: React.ReactElement }} [components] - React elements to use for interpolation
+ * @property {number} [count] - Count value for pluralization
+ * @property {string} [defaults] - Default text if translation is not found
+ * @property {string} [ns] - Namespace for the translation key
+ * @property {boolean} [shouldUnescape] - Whether to unescape HTML entities
+ * @property {Record<string, unknown>} [values] - Values to interpolate into the translation
```

Comment:
Nit: hmm, I am wondering if repeating the type definitions in the docs as well could cause any discrepancies in the future. Do we have any automatic validation for these (if they drift)? Would the types for the props be inferred if we would remove them and only leave the description? 🤔 

(Another question - would TSDoc / JSDoc behave the same if we would add the description in a `/** ... */` doc-block above the properties? I am unfortunately not sure if we have any best-practices regarding this.)

Reply:
Good catch, I'll remove those 👍

Reply:
Fixed 👍